proc/proc.go

```diff
@@ -527,6 +545,9 @@ func (dbp *Process) SwitchGoroutine(gid int) error {
 // GoroutinesInfo returns an array of G structures representing the information
 // Delve cares about from the internal runtime G structure.
 func (dbp *Process) GoroutinesInfo() ([]*G, error) {
+	if dbp.exited {
+		return []*G{}, nil
```

Comment:
Let's do it for both, since that error message will be more explicit and give the user some more info.